mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java

```diff
@@ -0,0 +1,61 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+
+package org.apache.james.transport.matchers;
+
+import com.google.common.collect.ImmutableSet;
+import org.apache.james.core.MailAddress;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMatcher;
+
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import java.util.*;
+
+/**
+ * <p>This matcher checks if the content type matches.</p>
+ */
+public class HasMimeType extends GenericMatcher {
+
+    private Set<String> content_types;
+
+    public void init() throws javax.mail.MessagingException {
+        content_types = ImmutableSet.copyOf(getCondition().split("\\s*,\\s*"));
+    }
+
+    public Collection<MailAddress> match(Mail mail) throws javax.mail.MessagingException {
+        String mimeType;
+        try {
```

Comment:
You can move the try/catch in getMimeTypeFrom message.

It avoid mimeType double allocation in parallel branches.